lib/event_grid.jl

```diff
@@ -0,0 +1,75 @@
+
+###########################################
+## Interpolated event grid (requires Grid)
+###########################################
+
+@comment """
+# Interpolated event grid
+"""
+
+using Requires
+
+@require Grid begin
+
+    using Grid
+    
+    function search_grid (g::InterpGrid, x::Float64)
+        i = searchsortedfirst (g, x)
+        g[i]
+    end
+
+    function grid_point (x::Float64, v::Float64)
+        x - v
+    end
+    
+@doc* """
+Interpolated event grid (requires Grid library).
+
+```julia
+
+EventGrid(events, tf, dt)
+```
+
+Given a vector with event times and a sampling step, creates a
+continuous interpolated function that has a negative value between
+events, and crosses zero at each event time.
+The function `grid_input` can be used to detect events in model equations.
+
+### Arguments
+
+* `event` : Vector of event times
+* `tf` : grid end time
+* `dt` : event grid sampling time
+
+### Example
+
+```julia
+
+input = EventGrid([1.0,2.0,3.0],tstop,0.1)
```

Comment:
Thanks, Ivan. Now for another naive question: could you merge EventGrid and grid_input? Then, it'd look like:

```julia
Event(eventgrid([1.0,2.0,3.0],tstop,0.1), ...
```
I'm not sure if these are ever needed independently.

Also, on capitalization, I think these should be lower case. They are not types or models.

Lastly, can tf and dt have defaults, possibly based on the range and smallest differences of the event times?